libs/gl-client/src/lnurl/pay/models.rs

```diff
@@ -0,0 +1,88 @@
+use crate::lnurl::tor_http_client::TorHttpClient;
+use anyhow::{anyhow, Result};
+use mockall::automock;
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+use tokio::runtime::Runtime;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct PayRequestResponse {
+    pub callback: String,
+    #[serde(rename = "maxSendable")]
+    pub max_sendable: u64,
+    #[serde(rename = "minSendable")]
+    pub min_sendable: u64,
+    pub tag: String,
+    pub metadata: String,
+}
+
+#[derive(Deserialize)]
+pub struct PayRequestCallbackResponse {
+    pub pr: String,
+    pub routes: Vec<String>,
+}
+
+#[automock]
+pub trait LnUrlHttpClient {
+    fn get_pay_request_response(&self, lnurl: &str) -> Result<PayRequestResponse>;
+    fn get_pay_request_callback_response(
+        &self,
+        callback_url: &str,
+        amount: u64,
+    ) -> Result<PayRequestCallbackResponse>;
+}
+
+pub struct LnUrlHttpClientImpl {
+    clearnet_client: reqwest::Client,
+    tor_client: TorHttpClient,
+}
+
+impl LnUrlHttpClientImpl {
+    /// Constructs a new `LnUrlHttpClient`.
+    ///
+    /// # Panics
+    ///
+    /// This method panics if a http or tor client cannot be initialized.
+    pub fn new() -> LnUrlHttpClientImpl {
+        let tokio_runtime = Runtime::new().expect("Failed to create Tokio runtime");
+        let tor_client = tokio_runtime.block_on(TorHttpClient::new()).unwrap();
+        LnUrlHttpClientImpl {
+            clearnet_client: reqwest::Client::new(),
+            tor_client,
+        }
+    }
+
+    fn get<T: DeserializeOwned + 'static>(&self, url: &str) -> Result<T> {
```

Comment:
If we make this `async` we do not have to build a nested `Runtime`.

Reply:
The goal here was not to make a full http client. There's no need for the use case of resolving lnurls. Maybe naming it `TorHttpClient` gave the wrong impression of the intended goal. 

All the examples of connecting a reqwest http client over Tor uses proxies. The arti crate [supports running a socks proxy](https://docs.rs/arti/latest/arti/#arti) but there were less examples for that so I opted to go for the higher-level arti-client crate which showed the client making [manual http requests in the docs](https://docs.rs/arti-client/0.10.0/arti_client/#example-making-connections-over-tor).

I'll defer the discussion to a later PR where I'll re-include the now-removed `tor_http_client`.